cmd/imgcollector/collector/collector.go

```diff
@@ -157,7 +157,21 @@ func (c *Collector) getImage(ctx context.Context) (image.ImageWithIndex, func(),
 	}
 	if c.cfg.Mode == config.ModeRemote {
 		opts := image.DockerOption{}
-		if c.cfg.DockerOptionPath != "" {
+		if c.cfg.ImagePullSecret != "" {
+			configData, err := config.ReadImagePullSecret(os.DirFS(config.SecretMountPath))
+			if err != nil {
+				return nil, nil, fmt.Errorf("reading image pull secret: %w", err)
+			}
+			cfg := image.DockerConfig{}
+			if err := json.Unmarshal(configData, &cfg); err != nil {
+				return nil, nil, fmt.Errorf("parsing image pull secret: %w", err)
+			}
+			if auth, ok := cfg.Auths[imgRef.Context().Registry.Name()]; ok {
```

Comment:
```json
"auths": {
		"ghcr.io": {},
		"https://index.docker.io/v1/": {},
		"registry.gitlab.com": {}
	}
```

This is default way to configure locally. But we should allow to specify registry name as this:

```
"ghcr.io": {},
"ghcr.io/my/private/registry": {}
```

This means man will not work here.

Reply:
Do I understand correctly that you're saying to create a custom image pull secret format instead of using the standard one? https://kubernetes.io/docs/tasks/configure-pod-container/pull-image-private-registry/
I was imagining that during the Service Account approach implementation, the collector would be told just a name of an image pull secret and that's it 😕 There would be no need in transforming secrets into our custom format.

Reply:
Is this what you had in mind?